### PR TITLE
Twilio sms

### PR DESCRIPTION
**Pull request by fleetwoods**

### CODE REVIEW

File: src/main/scala/com/ovoenergy/event/gateway/manager/Main.scala [hunk 1]
```diff
@@ -37,23 +50,72 @@ object Main extends App {
...
+    Producer.send[SMSProgressed](kafkaConfig, config.getString("kafka.topic.sms_progressed"))
+
+  val sendCommProgressedEvent = (issuedForDelivery: IssuedForDelivery) => {
+    issuedForDelivery.channel match {
+      case SMS   => sendSmsProgressedEvent(twilio.Converters.toSMSProgressed(issuedForDelivery))
+      case Email =>
+        logDebug(issuedForDelivery.metadata.traceToken,
+          s"No comm progressed (Accepted) event issued for $issuedForDelivery, these events still being handled in Delivery Service")
+        Future.successful(())
+      case _ =>
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/event/gateway/manager/Main.scala]: Not needed, compile time warning will happen

File: src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/TwilioResource.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    getGatewayIdentifier: (GatewayIdentifier.Id) => Option[GatewayIdentifier],
+    sendSmsProgressedEvent: (SMSProgressed) => Future[_])(implicit ec: ExecutionContext, mat: Materializer)
+    extends Directives
+    with LoggingWithMDC {
+
+  private def validateSignature: Directive0 = {
+    (extractRequest & formFieldMap).tflatMap {
+      case (request, formData) =>
+        request.headers.find(_.is("x-twilio-signature")) match {
+          case None => complete(Forbidden -> "Bad request!")
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/event/gateway/manager/api/resources/TwilioResource.scala]: Add some logging around these auth failures

File: src/main/scala/com/ovoenergy/event/gateway/manager/kafka/Producer.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    (event: T) =>
+      {
+        Retry.retryAsync(
+          config = config.retry,
+          onFailure = e => logWarn(event.metadata.traceToken, s"Failed to send Kafka event to topic $topic", e)
+        ) { () =>
+          producer
+            .send(new ProducerRecord[String, T](topic, event.metadata.customerId, event))
+            .map(record => {
+              logInfo(event.metadata.traceToken, s"Posted event to $topic - $event")
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/event/gateway/manager/kafka/Producer.scala]: Just log metadata not entire event

File: src/main/scala/com/ovoenergy/event/gateway/manager/twilio/Converters.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import com.ovoenergy.event.gateway.manager.dynamo.GatewayIdentifier
+
+object Converters {
+
+  private val statusMappings: Map[String, SMSStatus] = Map(
+    "queued"      -> Queued,
+    "failed"      -> Failed,
+    "sent"        -> Sent,
+    "accepted"    -> Accepted,
+    "sending"     -> Accepted,
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/event/gateway/manager/twilio/Converters.scala]: Make this Queued

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 107](https://circleci.com/gh/ovotech/comms-gateway-event-manager/107) at 2017-04-03 11:24 BST.

**Pull request merged**